content/doc/book/using/best-practices.adoc

```diff
@@ -0,0 +1,177 @@
+
+=== Use Pipeline
+
+If something prevents you from using organization folders and multibranch Pipelines, use Pipeline jobs.
+Organization folders and multibranch Pipelines are preferred over Pipeline jobs because they will automatically create and delete multibranch projects when repositories are created or deleted.
+
+Refer to the link:/doc/book/pipeline/[Pipeline] documentation for more details.
+
+.Differences between Freestyle and Pipeline in Jenkins
+video::IOUm1lw7F58[youtube,width=800,height=420]
+
+== Manage your jobs
+
+Jenkins job definitions can be managed and improved to make user interactions smoother and more productive.
+
+=== Report build results
+
+Charts and graphs show trends and patterns that can highlight project status and progress.
+Automated test results for unit tests, integration tests, and end-to-end tests can highlight brittleness or instability.
+Coverage reports can illustrate areas where automated tests are not executing.
+Compiler warning messages can be the first indication of a problem.
+Static analysis tools can report risky code or code that may have security risks.
+Performance test results can identify delays or problem areas.
+
+The plugin:warnings-ng[Warnings Next Generation] plugin provides convenient access to many reports, including:
+
+* Compiler warnnings and errors (like gcc, clang, javac, or  golang)
+* Static analysis warnings and errors (spotbugs, checkstyle, pmd, lint, cpd, or Simian)
+* Code coverage reports
+
+.How to use the Warnings Next Generation plugin
+video::tj3xYFA6Q2o[youtube,width=800,height=420]
+
+=== Build on agents
+
+Use agents to perform builds instead of running builds on the controller.
+It is much safer and more scalable to use agents.
+
+More details are available in the link:/doc/book/security/controller-isolation/[controller isolation] documentation.
+
+=== Show failures to the right people
+
+Configure the notifications from failing and unstable jobs so the notifications reach the right people without distracting others.
+Many Jenkins users prefer to only be notified when a failure is likely their fault.
+They correctly assume that if they did not cause the failure, they are not the best person to investigate the failure.
+
+Refine your notification system so that new test failures first notify the most recent committers, since they are the most likely cause of the issue.
+
+.Sending Slack notifications
+video::EDVZli8GdUM[youtube,width=800,height=420]
+
+=== Use simple project names
+
+Jenkins uses project names for folders related to the project.
+Many tools cannot handle spaces, dollar signs, or similar characters in file paths.
+Limit yourself to alphanumeric characters [a-zA-Z0-9_-]+ in project names.
+Use the *Display Name* feature to make them look nice.
+The "Restrict project naming" setting in the system configuration can enforce naming restrictions for all projects.
```

Comment:
```suggestion
To enforce consistent naming conventions across all projects, you can enable the "Restrict project naming" setting in the system configuration. This will ensure that naming restrictions are enforced uniformly.
```